### ISSUE TITLE
install requires gitpython and pyyaml before dependencies are calculated

### ISSUE THREAD

**Issue by BadgerOps** [iamseth/saltstack-vulcan#2]
The setup.py imports vulcan, which also imports vulcan.formula via the __init__.py - gitpython is a dependency of vulcan.formula, so the import fails. 

pyyaml is imported by vulcan.config which is also imported in the __init__.py and this import also fails

This prevents a 'fresh' install of saltstack-vulcan if gitpython isn't already installed, but will succeed as expected if gitpython and pyyaml are pre-installed.

example in venv below:

```bash
Variks:~ badger$ source saltstack-vulcan/bin/activate
(saltstack-vulcan) Variks:~ badger$ pip install saltstack-vulcan -U
Collecting saltstack-vulcan
  Using cached saltstack-vulcan-0.1.2.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/j9/j60g11dx0nbdh34lyqcgdy5c0000gn/T/pip-build-0xONLC/saltstack-vulcan/setup.py", line 3, in <module>
        import vulcan
      File "vulcan/__init__.py", line 5, in <module>
        import vulcan.formula
      File "vulcan/formula.py", line 7, in <module>
        import git
    ImportError: No module named git

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/j9/j60g11dx0nbdh34lyqcgdy5c0000gn/T/pip-build-0xONLC/saltstack-vulcan/
(saltstack-vulcan) Variks:~ badger$ pip install gitpython
Collecting gitpython
  Downloading GitPython-2.1.3-py2.py3-none-any.whl (442kB)
    100% |████████████████████████████████| 450kB 2.0MB/s
Collecting gitdb2>=2.0.0 (from gitpython)
  Downloading gitdb2-2.0.0-py2.py3-none-any.whl (63kB)
    100% |████████████████████████████████| 71kB 5.4MB/s
Collecting smmap2>=2.0.0 (from gitdb2>=2.0.0->gitpython)
  Downloading smmap2-2.0.1-py2.py3-none-any.whl
Installing collected packages: smmap2, gitdb2, gitpython
Successfully installed gitdb2-2.0.0 gitpython-2.1.3 smmap2-2.0.1
(saltstack-vulcan) Variks:~ badger$ pip install saltstack-vulcan -U
Collecting saltstack-vulcan
  Using cached saltstack-vulcan-0.1.2.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/j9/j60g11dx0nbdh34lyqcgdy5c0000gn/T/pip-build-giOwAP/saltstack-vulcan/setup.py", line 3, in <module>
        import vulcan
      File "vulcan/__init__.py", line 6, in <module>
        import vulcan.config
      File "vulcan/config.py", line 5, in <module>
        import yaml
    ImportError: No module named yaml

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/j9/j60g11dx0nbdh34lyqcgdy5c0000gn/T/pip-build-giOwAP/saltstack-vulcan/
(saltstack-vulcan) Variks:~ badger$ pip install pyyaml
Collecting pyyaml
  Downloading PyYAML-3.12.tar.gz (253kB)
    100% |████████████████████████████████| 256kB 1.2MB/s
Building wheels for collected packages: pyyaml
  Running setup.py bdist_wheel for pyyaml ... done
  Stored in directory: /Users/badger/Library/Caches/pip/wheels/2c/f7/79/13f3a12cd723892437c0cfbde1230ab4d82947ff7b3839a4fc
Successfully built pyyaml
Installing collected packages: pyyaml
Successfully installed pyyaml-3.12
(saltstack-vulcan) Variks:~ badger$ pip install saltstack-vulcan -U
Collecting saltstack-vulcan
  Using cached saltstack-vulcan-0.1.2.tar.gz
Requirement already up-to-date: GitPython>=2.1.3 in ./saltstack-vulcan/lib/python2.7/site-packages (from saltstack-vulcan)
Requirement already up-to-date: PyYAML>=3.12 in ./saltstack-vulcan/lib/python2.7/site-packages (from saltstack-vulcan)
Requirement already up-to-date: gitdb2>=2.0.0 in ./saltstack-vulcan/lib/python2.7/site-packages (from GitPython>=2.1.3->saltstack-vulcan)
Requirement already up-to-date: smmap2>=2.0.0 in ./saltstack-vulcan/lib/python2.7/site-packages (from gitdb2>=2.0.0->GitPython>=2.1.3->saltstack-vulcan)
Building wheels for collected packages: saltstack-vulcan
  Running setup.py bdist_wheel for saltstack-vulcan ... done
  Stored in directory: /Users/badger/Library/Caches/pip/wheels/85/97/bb/929efab164c65d0fa2b9f198c8dd9e9767a2a5faf6b3ffcebe
Successfully built saltstack-vulcan
Installing collected packages: saltstack-vulcan
Successfully installed saltstack-vulcan-0.1.2
(saltstack-vulcan) Variks:~ badger$
```

**Comment by BadgerOps** [comment]
This can be resolved with https://github.com/iamseth/saltstack-vulcan/pull/3

**Comment by iamseth** [comment]
Thanks for this. The install was definitely broken. I fixed in e116b5d82463d9d6b48a75eee5ec299dfc457444 to avoid duplicating the version number.

I think the logging stuff is still good.

**Comment by BadgerOps** [comment]
Thanks, updated #3
